driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -164,6 +164,11 @@ public void run() {
                             PooledConnection connection = getPooledConnection(getRemainingWaitTime(), MILLISECONDS);
                             openAsync(connection, errHandlingCallback);
                         }
+                    } catch (MongoTimeoutException e) {
+                        MongoTimeoutException exception = new MongoTimeoutException(format("Timeout waiting for a pooled item after %d %s",
```

Comment:
Might as well say "pooled connection" rather than "pooled item".

Reply:
Want me to change that down stream as well?

Reply:
Actually in the Concurrent Pool its generic. So won't change there.